### ISSUE TITLE
mx.metric.Perplexity should support calculating from cross-entropy

### ISSUE THREAD

**Issue by szha** [apache/incubator-mxnet#9590]
Currently part of the logic for calculating cross-entropy loss is in mx.metric.Perplexity. This means we would have repeated calculation when we are already using cross entropy loss, which is very commonly used (e.g SoftmaxCrossEntropyLoss). In that case, only exponentiating is needed.

https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/metric.py#L648-L658
